rust/core-lib/src/editor.rs

```diff
@@ -762,6 +762,83 @@ impl<W: Write + Send + 'static> Editor<W> {
         self.insert(&*String::from(kill_ring_string));
     }
 
+    pub fn do_find(&mut self, chars: Option<&str>, case_sensitive: bool) {
+        use xi_rope::find::{find, CaseMatching};
+
+        let search_string = chars.map(Cow::Borrowed).or_else(|| {
+            self.view.sel_regions().iter().last().and_then(|region| {
+                if region.is_caret() {
+                    None
+                } else {
+                    Some(Cow::Owned(self.text.slice_to_string(region.min(), region.max())))
+                }
+            })
+        });
+
+        if search_string.is_none() {
+            return;
+        }
+
+        let search_string = search_string.unwrap();
+        if search_string.len() == 0 {
+            return
+        }
+
+        let case_matching = if case_sensitive {
+            CaseMatching::Exact
+        } else {
+            CaseMatching::CaseInsensitive
+        };
+
+        let mut sb = SpansBuilder::new(self.text.len());
+        let len = search_string.len();
+
+        let mut cursor = Cursor::new(&self.text, 0);
+        while let Some(pos) = find(&mut cursor, case_matching, &search_string) {
+            let style = Style { fg: 0xffc00000, font_style: 0 };
+            sb.add_span(Interval::new_closed_open(pos, pos + len), style);
+        }
+
+        let spans = sb.build();
+        if spans.len() > 0 {
+            self.style_spans = spans;
+            self.view.set_dirty();
+        }
+    }
+
+    fn do_find_next(&mut self, reverse: bool, wrapped: bool) {
+        use styles::SEARCH_STYLE_ID;
+
+        let iv = self.view.sel_regions().iter().last().map(|region| {
+            if reverse != wrapped {
+                Interval::new_closed_open(0, region.min())
+            } else {
+                Interval::new_closed_open(region.max(), self.text.len())
+            }
+        });
+        if let Some(iv) = iv {
+            let offset = iv.start();
+            let spans = self.style_spans.subseq(iv);
+            let mut iter: Box<Iterator<Item=(Interval, &Style)>> = if reverse {
+                Box::new(spans.iter().rev())
+            } else {
+                Box::new(spans.iter())
+            };
+            let span = iter.find(|&(_, _style)| {
+                let style_id = SEARCH_STYLE_ID; // self.doc_ctx.get_style_id(style);
```

Comment:
Instead of using `self.doc_ctx.get_style_id(style);`, this still accepts all spans as a search highlight. Just wanted to highlight this issue again, even through I personally see this being fixed in later PRs (that e.g. focus on span-layers).